### ISSUE TITLE
Won't upload new or modified notes from multiple tablets

### ISSUE THREAD

**Issue by rmcconne** [nextcloud/notes#437]
I have Nextcloud 17.0.2 running on a Slackware64-current server on my home network. I have two tablets and two phones (all Android) with the latest Notes installed from the F-Droid repository. Only one of each still have an active data plan with Verizon. All are configured to upload only via WiFi and only when connected to my WAP, both bands of which are configured with SSID broadcast turned off. There is a m0n0wall fork running on a dual-port computer between my networks and Roadrunner. Updates and new notes are made on whichever device is easiest to reach. Occasionally edits and new notes are entered while out of WiFi range.
After editing or creating a note, some will sit and spin attempting to upload, but even two days later will not have uploaded.
Phones: Pixel 3a, Samsung S6
Tablets: ASUS Zenpad Z8, Samsung Galaxy Tab E
WAP: Netgear R6020 Dual-band, uses firewall for DNS. FIrewall is configured to accept just the machine name for internal lookup only.
Steps to secure WAP: 
   1. Turn off SSID broadcast.
   2. Change SSID to be at least 24 characters.
   3. Add password of at least 36 characters.

**Comment by stefan-niedermann** [comment]
Please provide some more information:

- Did you upgrade the Notes android app recently or fresh install?
- Which version was the last that worked?
- Does the files app work well?
- Do you see any error message (which?)?
- What are the exact versions of the files app and the notes app?
- Do the server logs hint to an inconing call or is not even a trace of a request visible?

**Comment by stefan-niedermann** [comment]
> All are configured to upload [...] only when connected to my WAP

How? The notes app does only provide an option to limit syncs to when wi-fi is connected.

**Comment by rmcconne** [comment]
On 1/17/20 11:54 AM, Niedermann IT-Dienstleistungen wrote:
> Please provide some more information:
> 
> - Did you upgrade the Notes android app recently or fresh install?
> - Which version was the last that worked?
> - Does the files app work well?
> - Do you see any error message (which?)?
> - What are the exact versions of the files app and the notes app?
> - Do the server logs hint to an inconing call or is not even a trace of a request visible?
> 

Notes has been upgraded several times via F-Droid.
I don't know that it ever worked.
What files app? All I have is the Nextcloud Syncronization client 
version 3.9.2. I don't see any "Files" app on F-Droid.
There are no errors, nor status messages that I can find.
Notes is currently at 2.4.4.
On the server, error log only shows the shutdown and startup messages 
when logrotate ran each morning. the access.log has more than 4500 lines 
in the last 11 hours, what specifically am I looking for?

-- 
Bob McConnell
N2SPP

**Comment by rmcconne** [comment]
On 1/17/20 11:58 AM, Niedermann IT-Dienstleistungen wrote:
>> All are configured to upload [...] only when connected to my WAP
> 
> How? The notes app does only provide an option to limit syncs to when wi-fi is connected.
> 

Ok, that one is in the DAVx5 settings. Too bad you can't share those.

-- 
Bob McConnell
N2SPP

**Comment by stefan-niedermann** [comment]
> All I have is the Nextcloud Syncronization client

Yes, that is what i mean :) Does it work?

Please, first of all, update the Notes android app to at least 2.4.6, where the exception handling has been improved and we hopefully see more information about why the sync fails.

Also: Is it possible for you to get a Logcat? There are apps on F-Droid for catching logcats.

**Comment by stefan-niedermann** [comment]
Regarding the server logs: I do not know exactly what we are looking for, but you can have a look at the clock, perform some manual refreshs (swipe down i the notes list) and then have a look at what has been logged with this time stamp.

**Comment by rmcconne** [comment]
On 1/18/20 1:43 AM, Niedermann IT-Dienstleistungen wrote:
> Regarding the server logs: I do not know exactly what we are looking for, but you can have a look at the clock, perform some manual refreshs (swipe down i the notes list) and then have a look at what has been logged with this time stamp.
> 

There is one detail I noted from my first review of those logs. The 
server is on an Ethernet segment, while the mobile devices are all on 
WiFi. Because the router is doing NAT, all of those connections appear 
to be from the same IP address, that of the router's WAN port. I don't 
know what NextCloud is using to keep track of the individual devices, so 
don't know if that makes any difference.

-- 
Bob McConnell
N2SPP

**Comment by kevinkk525** [comment]
This might be the same problem as #350 ?

**Comment by stefan-niedermann** [comment]
Maybe. @rmcconne can you please tell me if you are using "Ransomware Recovery"-app on your Nextcloud instance?

**Comment by rmcconne** [comment]
On 2/5/20 4:24 PM, Niedermann IT-Dienstleistungen wrote:
> Maybe. @rmcconne can you please tell me if you are using "Ransomware Recovery"-app on your Nextcloud instance?
> 

No, I have none of that. I'm not even sure that would run since the 
server is on Slackware Linux.

I do have a number of error logs from Notes now on my Pixel 3A. I just 
have to figure out how to move them over here where I can email them.

-- 
Bob McConnell
N2SPP

**Comment by stefan-niedermann** [comment]
info@niedermann.it

**Comment by korelstar** [comment]
Any news on this?

**Comment by rmcconne** [comment]
Good morning,

I have a couple of issues that someone needs to think about. Earlier 
this week I attempted to update Nextcloud to 18.0.2, but it failed on my 
Slackware64 server. Therefore, I downloaded the zip file and reinstalled 
it. But when I opened Notes on the phone, instead of uploading the 
existing notes onto the clean server, it wiped them out. I ran into a 
similar issue when I updated to 18.0.1, where several updates made to 
Notes on the phone while the server was indisposed were removed when it 
finally came back. The app really needs to ask which notes or versions 
to keep before it erases them.

I did try to find the files on the phone before I did the update, to 
back them up in case this happened, but was unable to locate them. Where 
are they hidden?

Is there any way to restore the files on the phone, or are they gone?

I do have a two day old backup from the server which will not be easy to 
restore, but that still loses several updates that may not be possible 
to recreate.

One final comment about updates. After I open a note just to read it, 
the app still appears to upload it to the server as if I had edited it. 
Why doesn't the app know that I didn't make any changes to the file? 
This is particularly irritating when I am out of WiFi range.

Both the Pixel 3A and the Samsung S6 behave the same. I haven't tried 
the tablets yet.

-- 
Bob McConnell
N2SPP

**Comment by korelstar** [comment]
> Good morning, I have a couple of issues that someone needs to think about. Earlier this week I attempted to update Nextcloud to 18.0.2, but it failed on my Slackware64 server. Therefore, I downloaded the zip file and reinstalled it. But when I opened Notes on the phone, instead of uploading the existing notes onto the clean server, it wiped them out. I ran into a similar issue when I updated to 18.0.1, where several updates made to Notes on the phone while the server was indisposed were removed when it finally came back. The app really needs to ask which notes or versions to keep before it erases them.

This is not a bug. When updating the server, you have to keep the data directory on the server. If you re-install Nextcloud with a clean data directory (without copying the old data), the client must assume that the notes were deleted intentionally on the server (the client can't know, whether there were an update on the server or the notes were manually deleted). Consequently, all notes are deleted on the phone, too.

Hence, you always have to load a backup if you are doing a complete re-installation. However, this is a typical procedure in every server software.

> I did try to find the files on the phone before I did the update, to back them up in case this happened, but was unable to locate them. Where are they hidden? Is there any way to restore the files on the phone, or are they gone? I do have a two day old backup from the server which will not be easy to restore, but that still loses several updates that may not be possible to recreate.

In the Android app, notes are stored in a sqlite database (like most data from Android apps). You can backup and restore it using respective Android backup tools. However, you would ran into the same issue and you will lose your notes again after the next synchronization. Therefore, you have to do server-side backups instead of/additional to client-side backups.

> One final comment about updates. After I open a note just to read it, the app still appears to upload it to the server as if I had edited it. Why doesn't the app know that I didn't make any changes to the file? This is particularly irritating when I am out of WiFi range. Both the Pixel 3A and the Samsung S6 behave the same. I haven't tried the tablets yet.

This should not happen. Please open a new issue for this in the correct repository (https://github.com/stefan-niedermann/nextcloud-notes) as this is different from the original topic.

Regarding the original problem in this issue: does the problem still happens or can we close this issue?

**Comment by rmcconne** [comment]
On 3/14/20 3:48 PM, korelstar wrote:
>> Good morning, I have a couple of issues that someone needs to think about. Earlier this week I attempted to update Nextcloud to 18.0.2, but it failed on my Slackware64 server. Therefore, I downloaded the zip file and reinstalled it. But when I opened Notes on the phone, instead of uploading the existing notes onto the clean server, it wiped them out. I ran into a similar issue when I updated to 18.0.1, where several updates made to Notes on the phone while the server was indisposed were removed when it finally came back. The app really needs to ask which notes or versions to keep before it erases them.
> 
> This is not a bug. When updating the server, you have to keep the data directory on the server. If you re-install Nextcloud with a clean data directory (without copying the old data), the client must assume that the notes were deleted intentionally on the server (the client can't know, whether there were an update on the server or the notes were manually deleted). Consequently, all notes are deleted on the phone, too.
> 
> Hence, you always have to load a backup if you are doing a complete re-installation. However, this is a typical procedure in every server software.
> 
>> I did try to find the files on the phone before I did the update, to back them up in case this happened, but was unable to locate them. Where are they hidden? Is there any way to restore the files on the phone, or are they gone? I do have a two day old backup from the server which will not be easy to restore, but that still loses several updates that may not be possible to recreate.
> 
> In the Android app, notes are stored in a sqlite database (like most data from Android apps). You can backup and restore it using respective Android backup tools. However, you would ran into the same issue and you will lose your notes again after the next synchronization. Therefore, you have to do server-side backups instead of/additional to client-side backups.
> 
>> One final comment about updates. After I open a note just to read it, the app still appears to upload it to the server as if I had edited it. Why doesn't the app know that I didn't make any changes to the file? This is particularly irritating when I am out of WiFi range. Both the Pixel 3A and the Samsung S6 behave the same. I haven't tried the tablets yet.
> 
> This should not happen. Please open a new issue for this in the correct repository (https://github.com/stefan-niedermann/nextcloud-notes) as this is different from the original topic.
> 
> Regarding the original problem in this issue: does the problem still happens or can we close this issue?
> 

I have been able to update from multiple devices with no issues for the 
past two or three versions now. So I believe you can close that one.

Now I just have to figure out how to restore the Nextcloud database from 
18.0.1 onto 18.0.2. I tried recreating them using the Notes app on the 
web page, but the Android app doesn't recognize those files.

-- 
Bob McConnell
N2SPP

**Comment by stefan-niedermann** [comment]
Did you run occ:scan comand?

**Comment by rmcconne** [comment]
On 3/14/20 6:43 PM, Niedermann IT-Dienstleistungen wrote:
> Did you run occ:scan comand?
> 

Command not found!

-- 
Bob McConnell
N2SPP

**Comment by rmcconne** [comment]
On 3/14/20 6:43 PM, Niedermann IT-Dienstleistungen wrote:
> Did you run occ:scan comand?
> 

Ok, this line apparently did the trick:

   $ sudo -u apache php occ files:scan

Then I had to close the Notes app through the multi-app display on the 
phone and restart it before it would sync. Now I can see those notes on 
the phone.

Thank you,

-- 
Bob McConnell
N2SPP

**Comment by stefan-niedermann** [comment]
You're welcome. You should (if possible) never copy files to the folders directly using FTP or so. Always go via sync client or webdav. If you really *need* to directly write files inside of the nextcloud data folder, you will have to run this command to notify Nextcloud to update it's database.

**Comment by rmcconne** [comment]
On 3/14/20 8:17 PM, Niedermann IT-Dienstleistungen wrote:
> You're welcome. You should (if possible) never copy files to the folders directly using FTP or so. Always go via sync client or webdav. If you really *need* to directly write files inside of the nextcloud data folder, you will have to run this command to notify Nextcloud to update it's database.
> 

I hope this was a unique circumstance. The 18.0.1 to 18.0.2 upgrade 
failed with a 'directory not found' error when I selected either the 
update or download button, so I had to do a clean install. Then I copied 
the Notes and Photo directories from the copy of 18.0.1 that I did just 
before that install. Fortunately, those are the only two features that I 
have made significant use of so far.

-- 
Bob McConnell
N2SPP
